src/passes/hash-stringify-walker.cpp

```diff
@@ -170,17 +172,19 @@ std::vector<SuffixTree::RepeatedSubstring> StringifyProcessor::filterOverlaps(
   // Get the overlapping intervals
   std::vector<SuffixTree::RepeatedSubstring> result;
   std::unordered_set<unsigned> substringsIncluded;
+  std::vector<Index> seenCount(substrings.size(), 0);
   std::vector<int> indices = IntervalProcessor::filterOverlaps(intervals);
   for (auto i : indices) {
     // i is the idx of the Interval in the intervals vector
     // i in substringIdxs returns the idx of the substring that needs to be
     // included in result
     auto substringIdx = substringIdxs[i];
-    if (substringsIncluded.find(substringIdx) != substringsIncluded.end()) {
-      continue;
+    seenCount[substringIdx] += 1;
+    auto& substring = substrings[substringIdx];
+    if (seenCount[substringIdx] == substring.StartIndices.size() &&
+        substringsIncluded.insert(substringIdx).second) {
+      result.push_back(substring);
```

Comment:
So we're only considering a substring for outlining at all if all of its ocurrences survive overlap filtering? Could we keep the substring in consideration and just remove the particular occurrence of it that had the overlap instead?

Reply:
done